docs/docs/development/hardware-integration/pinctrl.mdx

```diff
@@ -0,0 +1,345 @@
+---
+title: Board Pin Control
+sidebar_label: Pin Control
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+:::info
+This page exists to provide a guide to [Pin Control](https://docs.zephyrproject.org/3.5.0/hardware/pinctrl/index.html#pin-control) for ZMK users and designers. Refer to [Zephyr's page on Pin Control](https://docs.zephyrproject.org/3.5.0/hardware/pinctrl/index.html#pin-control) for elaboration and more details on any of the points raised here.
+:::
+
+A basic keyboard design as introduced in the [new shield guide](./new-shield.mdx) only uses its pins for the keyboard matrix. Many keyboard designs make use of advanced components or functionality, such as displays or shift registers. This results in the keyboard making use of communication protocols such as (but not limited to) SPI, I2C, or UART. Configuring pins for the usage of advanced functionality such as drivers for the previously named protocols is referred to as "Pin Control".
+
+:::warning
+The details of pin control can vary from vendor to vendor. An attempt was made to be as general as possible, but it isn't possible to cover all possible cases. The approaches for the nRF52840 and rp2040 MCUs/SoCs are documented in their entirety below. For other MCUs/SoCs, please refer to the [Zephyr documentation](https://docs.zephyrproject.org/3.5.0/index.html) and the examples and other files found in-tree of [ZMK](https://github.com/zmkfirmware/zmk/tree/main/app/boards) and [ZMK's fork of Zephyr](https://github.com/zmkfirmware/zephyr).
```

Comment:
I think these links also apply to finding an existing pinctrl node/bus for a given board, assuming the user is writing a shield. It might be good info, given e.g. Xiao buses are defined like you note below.

Reply:
While true, I think that would be overloading this block a bit. Not sure on best way to handle.

Reply:
I realized that these are actually documented for the defined interconnects at 
https://deploy-preview-2508--zmk.netlify.app/docs/development/hardware-integration/new-shield#shield-overlays already. This might be a good opportunity to copy those images here, along with the bus node names as documented. Maybe it can be a separate subsection like "Existing pinctrl definitions for interconnects" (draft name) that replaces the tip box right below?

Reply:
Good idea, have done so. Had to mess with the interconnect tabs bit somewhat, but it works.